lua/neogit/buffers/commit_select_view/ui.lua

```diff
@@ -1,43 +1,20 @@
-local Ui = require("neogit.lib.ui")
+local util = require("neogit.lib.util")
 
-local row = Ui.row
-local text = Ui.text
+local Commit = require("neogit.buffers.common").CommitEntry
+local Graph = require("neogit.buffers.common").CommitGraph
 
 local M = {}
 
-function M.View(commits, commit_at_cursor)
-  local show_graph = true
-
-  local res = {}
-  if commit_at_cursor then
-    table.insert(
-      res,
-      row {
-        text(commit_at_cursor.oid:sub(1, 7), { highlight = "Number" }),
-        text(" "),
-        text(show_graph and ("* "):rep(commit_at_cursor.level + 1) or "* ", { highlight = "Character" }),
-        text(" "),
-        text(table.concat(commit_at_cursor.description)),
-      }
-    )
-
-    table.insert(res, row {})
-  end
-
-  for _, commit in ipairs(commits) do
-    table.insert(
-      res,
-      row {
-        text(commit.oid:sub(1, 7), { highlight = "Number" }),
-        text(" "),
-        text(show_graph and ("* "):rep(commit.level + 1) or "* ", { highlight = "Character" }),
-        text(" "),
-        text(table.concat(commit.description)),
-      }
-    )
-  end
-
-  return res
+---@param commits CommitLogEntry[]
```

Comment:
Does this in some way retain the *commit under cursor* functionality?

What it did was if you had the cursor on the `Recent Commits` view or alike it would show that commit at the top so that you can navigate to a commit in the log and e.g invoke rebase from there without chasing your commit again.

Reply:
Yes! Two ways:
- if you have a commit(s) selected in visual mode, those will be selected automatically.
- otherwise its using `vim.fn.search` in the `after` callback to locate the commit.